### PR TITLE
[mongoose] Add missing overwriteDiscriminatorKey option

### PR DESCRIPTION
**Pull request by Oniurain**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: [here](https://mongoosejs.com/docs/api.html#query_Query-setOptions)
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

No need to change the version number in the header because the feature is here since [5.10](https://github.com/Automattic/mongoose/blob/master/History.md#5100--2020-08-14)

### CODE REVIEW

File: types/mongoose/test/model.ts [hunk 1]
```diff
@@ -230,7 +231,7 @@ MongoModel.deleteOne({_id: '999'}).exec().then(res=>console.log(res.ok));
 MongoModel.deleteMany({_id: '999'}).then(res=>console.log('Success?',!!res.ok, 'deleted count', res.n));
 MongoModel.deleteMany({_id: '999'}).exec().then(res=>console.log(res.ok));
 MongoModel.update({ age: { $gt: 18 } }, { oldEnough: true }, cb);
-MongoModel.update({ name: 'Tobi' }, { ferret: true }, { multi: true,  arrayFilters: [{ element: { $gte: 100 } }] }, cb);
+MongoModel.update({ name: 'Tobi' }, { ferret: true }, { multi: true,  arrayFilters: [{ element: { $gte: 100 } }], overwriteDiscriminatorKey: true }, cb);
```
Comment (1pete) [inline types/mongoose/test/model.ts:234]: Better not modify the existing line, but add this as a new case. To ensure that without this option, it still works.

Comment (Oniurain) [reply types/mongoose/test/model.ts:235]: Done

**Review by 1pete — APPROVED** [review]

**Comment by Oniurain** [comment]
Ready to merge.

**Comment by PierrickGa** [comment]
Oui belle merged !

**Pull request merged**